### ISSUE TITLE
word_count example fails

### ISSUE THREAD

**Issue by bbonenfant** [pachyderm/pachyderm#7240]
**What happened?**:
Attempted to run the `word_count` example and the `map` pipeline failed.

**What you expected to happen?**:
All the `scraper`, `map`, and `reduce` pipeline should succeed.

**How to reproduce it (as minimally and precisely as possible)?**:
```bash
cd examples/word_count
make wordcount
```

**Anything else we need to know?**:
The output from inspecting the failed `map` job:
```
ID: 9d09880c690c49a98011e3247150f628
Pipeline: map
Started: 2 minutes ago 
Duration: 7 seconds 
State: failure
Reason: duplicate path output by different datums (/a from fe2a351981fea3a041465f5b9a42aefe0909b89a04c52628654efe10ffec5427 and /a from ff293f48de8d76333ecabca586b6b75a6df59a6d2e028c941c3ed92ecde163af)
Processed: 2
Failed: 0
Skipped: 0
Recovered: 0
Total: 2
Data Downloaded: 109.9KiB
Data Uploaded: 4.539KiB
Download Time: Less than a second
Process Time: Less than a second
Upload Time: Less than a second
Datum Timeout: (duration: nil Duration)
Job Timeout: (duration: nil Duration)
Worker Status:
WORKER              JOB                 DATUM               STARTED             
Restarts: 0
ParallelismSpec: <nil>



Input:
{
  "pfs": {
    "name": "scraper",
    "repo": "scraper",
    "repo_type": "user",
    "branch": "master",
    "commit": "9d09880c690c49a98011e3247150f628",
    "glob": "/*/*"
  }
}

Transform:
{
  "image": "pachyderm/example-wordcount:1.12.2",
  "cmd": [
    "app",
    "/pfs/scraper/",
    "/pfs/out"
  ]
} 
Output Commit: 9d09880c690c49a98011e3247150f628 
```

**Environment?**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.3", GitCommit:"ca643a4d1f7bfe34773c74f79527be4afd95bf39", GitTreeState:"clean", BuildDate:"2021-07-15T21:04:39Z", GoVersion:"go1.16.6", Compiler:"gc", Platform:"darwin/arm64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.1", GitCommit:"5e58841cce77d4bc13713ad2b91fa0d961e69192", GitTreeState:"clean", BuildDate:"2021-05-21T23:01:33Z", GoVersion:"go1.16.4", Compiler:"gc", Platform:"linux/amd64"}
```
- Pachyderm CLI and pachd server version (use `pachctl version`):
```
COMPONENT           VERSION                                          
pachctl             2.0.5   
pachd               2.0.5  
```
- Cloud provider (e.g. aws, azure, gke) or local deployment (e.g. minikube vs dockerized k8s): `kind`
- If you deployed with helm, the values you used (`helm get values pachyderm`):
```
USER-SUPPLIED VALUES:
deployTarget: LOCAL
```
- OS (e.g. from /etc/os-release):  `Ubuntu 20.04.3 LTS`

**Comment by RaananHadar** [comment]
This looks like the example violates the [Single Datum Provenance Rule](https://docs.pachyderm.com/latest/concepts/pipeline-concepts/datum/relationship-between-datums/#4-final-commit-in-the-pipelines-output-repo).

**Comment by nadegepepin** [comment]
Fixed on Master in PR #7382 
Backport-ed to 2.0.x and 2.1.x in #7385 and #7386
